vpr/src/place/compressed_grid.h

```diff
@@ -6,38 +6,98 @@
 #include "vtr_geometry.h"
 #include "vtr_flat_map.h"
 
-struct t_type_loc {
-    int x = OPEN;
-    int y = OPEN;
-
-    t_type_loc(int x_val, int y_val)
-        : x(x_val)
-        , y(y_val) {}
-
-    //Returns true if this type location has valid x/y values
-    operator bool() const {
-        return !(x == OPEN || y == OPEN);
-    }
-};
-
 struct t_compressed_block_grid {
+    //TODO: compressed_to_grid_x/y should become a 2d vector
     //If 'cx' is an index in the compressed grid space, then
     //'compressed_to_grid_x[cx]' is the corresponding location in the
     //full (uncompressed) device grid.
-    std::vector<int> compressed_to_grid_x;
-    std::vector<int> compressed_to_grid_y;
+    std::vector<std::vector<int>> compressed_to_grid_x;
+    std::vector<std::vector<int>> compressed_to_grid_y;
+    std::vector<int> compressed_to_grid_layer;
```

Comment:
Comment "same for y and layer".  (Currently layers aren't compressed so it is an identity mapping).